src/GitHub.TeamFoundation.14/Services/VSGitExt.cs

```diff
@@ -41,52 +42,75 @@ public VSGitExt(IGitHubServiceProvider serviceProvider, IVSUIContextFactory fact
             this.repositoryFactory = repositoryFactory;
 
             // The IGitExt service isn't available when a TFS based solution is opened directly.
-            // It will become available when moving to a Git based solution (cause a UIContext event to fire).
+            // It will become available when moving to a Git based solution (and cause a UIContext event to fire).
             context = factory.GetUIContext(new Guid(Guids.GitSccProviderId));
 
-            // Start with empty array until we have a change to initialize.
+            // Start with empty array until we have a chance to initialize.
             ActiveRepositories = Array.Empty<ILocalRepositoryModel>();
 
-            if (context.IsActive && TryInitialize())
+            PendingTasks = InitializeAsync();
+        }
+
+        async Task InitializeAsync()
```

Comment:
We could explicitly call this after constructing the service.

Reply:
Since `IVSGitExt` is also consumed as a MEF service, we probably want the VS service to return as soon as possible (since there is no async loading for MEF services). Awaiting `InitializeAsync` before the service returns is probably a bad idea. Better to continue initializing in the background.